feature/dsql/token/auth_token_generator_test.go

```diff
@@ -0,0 +1,165 @@
+package token
+
+import (
+	"context"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/sdk"
+)
+
+type dbTokenTestCase struct {
+	endpoint            string
+	region              string
+	expires             time.Duration
+	credsExpireIn       time.Duration
+	expectedHost        string
+	expectedQueryParams []string
+	expectedError       string
+}
+
+type tokenGenFunc func(ctx context.Context, endpoint, region string, creds aws.CredentialsProvider, optFns ...func(options *AuthTokenOptions)) (string, error)
+
+func TestGenerateDbConnectAuthToken(t *testing.T) {
+	cases := map[string]dbTokenTestCase{
+		"no region": {
+			endpoint:      "https://prod-instance.us-east-1.rds.amazonaws.com:3306",
```

Comment:
Aurora DSQLs endpoints dont need to have schema (https / http) and the port information. Also the endpoints are dual stack only. An example format is this.
```suggestion
			endpoint:      "foo0bar1baz2quux3quuux4.dsql.us-east-1.on.aws",
```

Same goes for other examples below.

Reply:
Changed the endpoints to reflect this endpoint. I added them with scheme as well to make sure we handle this case as well and strip the scheme if passed by the developer